retrigger/triggerhandler.py

```diff
@@ -276,6 +276,34 @@ async def on_raw_message_edit(self, payload: discord.RawMessageUpdateEvent) -> N
             return
         await self.check_triggers(message, True)
 
+    @commands.Cog.listener()
+    async def on_message_edit(self, before_message: discord.Message, after_message: discord.Message) -> None:
+        guild = self.bot.get_guild(int(after_message.guild.id))
+        if not guild:
+            return
+        if guild.id not in self.triggers:
+            return
+        if await self.bot.cog_disabled_in_guild(self, guild):
+            return
+        if not any(t.embeds and t.check_edits for t in self.triggers[guild.id].values()):
+            # log.debug(f"No triggers in {guild=} have check_edits enabled")
+            return
+        if after_message.author.bot:
+            return
+        channel = guild.get_channel(int(after_message.channel.id))
+        try:
+            message = await channel.fetch_message(int(after_message.id))
+        except (discord.errors.Forbidden, discord.errors.NotFound):
+            log.debug(
+                "I don't have permission to read channel history or cannot find the message."
+            )
+            return
+        except Exception:
+            log.info("Could not find channel or message")
+            # If we can't find the channel ignore it
+            return
+        await self.check_triggers(after_message, True)
+
```

Comment:
`on_raw_message_edit` already exists, this will cause duplicate triggers having it in twice.

Reply:
I may have got something wrong in the logic here but on_raw_message_edit should only be triggering when it is `check_edits AND NOT embeds` meanwhile on_message_edit is `check_edits AND embeds`

Reply:
The problem with this is we're already fetching the message in `on_raw_message_edit` meaning every edited message is potentially pulling 2 of our 5/5s rate limit on message fetches which is not going to happen in this cog.  Here you just copied what already exists in `on_raw_message_edit` without even utilizing d.py internally already passing the new message to you in the event.

Reply:
I should clarify since I see what you mean now. The issue is your approach is not very DRY (Don't Repeat Yourself). We already have the logic for checking edits and we have a criteria that embeds only show up on an edited message so if both must be active we can utilize the already existing logic to get the message passed to the trigger checker. A trigger must have check edits enabled *and* read embeds in order to function so we're still filtering the edit fetch for any trigger that doesn't have check edits enabled, we don't need an extra check in there for embeds too. I would rather see a note that check edits is required for read embeds work properly than repeat the same code we have here since it is a requirement due to a discord limitation.

Reply:
That's okay I do fully understand what you mean, it was entirely my fault for not digging deeper when I ran into an issue!

I originally tested with raw but couldn't get a successful output and originally I was misreading the discord.py doco and thought it was doing something that was excluding embeds from the raw version. I have done some further digging and found the reason it didn't originally work was because I glazed over `if "content" not in payload.data`, not processing that raw only contains the edits, not the full message. This meant that when an embed was edited it doesn't have a `content` it has `embeds` and as such was returning on the very first check.

I have a current working version locally which I can push up soon to fix this all up and bring it into the one :)

Reply:
Removed extra listener and combined into `on_raw_message_edit`